common-items.lic

```diff
@@ -0,0 +1,55 @@
+# quiet
+=begin
+  Documentation: https://elanthipedia.play.net/mediawiki/index.php/Lich_script_development#common-items
+=end
+
+custom_require.call(%w(common common-travel))
+
+module DRCI
+  module_function
+
+  def buy_item(room, item)
+    DRCT.walk_to(room)
+
+    case DRC.bput("buy #{item}", /prepared to offer it to you for (.*) kronars/, /Let me but ask the humble sum of (.*) coins/, /for a mere (.*) kronars/, 'You decide to purchase')
+    when /prepared to offer it to you for (.*) kronars/
+      amount = Regexp.last_match(1)
+    when /Let me but ask the humble sum of (.*) coins/
+      amount = Regexp.last_match(1)
+    when /for a mere (.*) kronars/
+      amount = Regexp.last_match(1)
+    end
+
+    fput("offer #{amount}") if amount
+  end
+
+  def order_item(room, item_number)
+    DRCT.walk_to(room)
+
+    fput("order #{item_number}")
+    fput("order #{item_number}")
+  end
+
+  def dispose(item, trash_room = nil)
+    DRCT.walk_to(trash_room) unless trash_room.nil?
+    DRC.bput("get my #{item}", 'You get')
+    dispose_trash(item)
+  end
+
+  def dispose_trash(item)
+    return if item.nil?
+    trashcan = DRRoom.room_objs
+                     .map { |long_name| get_noun(long_name) }
+                     .find { |obj| $TRASH_STORAGE.include?(obj) }
+    if trashcan == 'gloop'
+      trashcan = 'bucket' if DRRoom.room_objs.include? 'bucket of viscous gloop'
+      trashcan = 'cauldron' if DRRoom.room_objs.include? 'small bubbling cauldron of viscous gloop'
```

Comment:
we could `.find` the room object with gloop and use a regex to pull out NOUN of viscous gloop.